### ISSUE TITLE
Upgrade web browser extension's manifest version t v3

### ISSUE THREAD

**Issue by BrunoBernardino** [padloc/padloc#510]
The manifest v2 (what we're currently using) will stop being supported in June 2023. We also can't publish to the Edge store with v2 anymore, and there is more info and context at https://github.com/padloc/padloc/issues/421#issuecomment-1196442418

**Comment by BrunoBernardino** [comment]
Wow. Did I go down on a rabbit hole on this one.

Basically, we can't keep the same functionality for the extension on v3 for neither Chrome (including Edge) nor Firefox. There are workarounds, but also many developers reporting those workarounds stop working on any given random release.

Even though we can get past the manifest/API implementations between Chrome and Firefox for v3, the main functionality we'll lose in both browsers when upgrading is to keep the app unlocked. Basically, the new manifest version makes the extension useless (in my opinion) unless we can somehow use biometric unlock with it, as every time the extension closes, it _might_ (unpredictable and at the discretion of the browser) need to be unlocked.

There's a lot of backlash from extension developers (and I remembered, reading some of the links I'll share below, hearing about it a few years ago), but it seems that while Firefox was amenable to implementing important differences, Google wasn't, and will have _many_ extensions, including the most popular ad-blocking ones, just stop working next year. I hope consumer and extension developer pressure will eventually make them concede on persistent background scripts, but we'll see.

In summary, I don't think it's worth pursuing this in the next 6 months _only_ to be able to upload to the Edge store, _while_ losing this important functionality. The extension can be installed and will continue to work from our releases page or Chrome's Web Store on Edge for at least another year, so Edge customers have that workaround.

Here are some relevant links to understand the current state of affairs:

- https://github.com/uBlockOrigin/uBlock-issues/issues/338
- https://stackoverflow.com/questions/66618136/persistent-service-worker-in-chrome-extension/66618269#66618269
- https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/
- https://extensionworkshop.com/documentation/develop/manifest-v3-migration-guide/
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension
- https://bugs.chromium.org/p/chromium/issues/detail?id=1152255
- https://discourse.mozilla.org/t/need-more-advice-on-non-persistence/102257
- https://stackoverflow.com/questions/66406672/chrome-extension-mv3-modularize-service-worker-js-file
- https://github.com/mozilla/webextension-polyfill/issues/329
- https://github.com/abhijithvijayan/web-extension-starter/issues/66
- https://www.eff.org/deeplinks/2021/12/googles-manifest-v3-still-hurts-privacy-security-innovation

**Comment by MaKleSoft** [comment]
Thanks for the detailed research, Bruno! I agree, let's table this for now.

**Comment by darkpixel** [comment]
Yup.  Apparently Google doesn't let you publish extensions with the v2 manifest anymore, but you can still update pre-existing v2 extensions.  That's not going to be much help for anyone though... ;)

![Screenshot from 2022-09-22 17-22-28](https://user-images.githubusercontent.com/158380/191872750-bb2a59a3-68be-48f9-82d7-4bc092cbf5f4.png)

**Comment by nhnifong** [comment]
`The extension can be installed and will continue to work from our releases page or Chrome's Web Store on Edge for at least another year, so Edge customers have that workaround.`

Chrome browsers managed by enterprises also disallow the V2 manifest extensions now by default unless the enterprise changes the policy
